iphone/cli/lib/process-js-task.js

```diff
@@ -0,0 +1,374 @@
+	}
+
+	/**
+	 * Processes a single JavaScript file by applying several transforms and then copying it to
+	 * its destination path.
+	 *
+	 * If the content of the unprocessed JavaScript file did not change compared to the last run,
+	 * any further processing will be skipped.
+	 *
+	 * @param {String} filePathAndName Full path to the JavaScript file
+	 * @return {Promise}
+	 */
+	processJsFile(filePathAndName) {
+		let file = this.resolveRelativePath(filePathAndName);
+		if (!file) {
+			throw new Error(`Unable to resolve relative path for ${filePathAndName}.`);
+		}
+
+		const bootstrapPath = file.substr(0, file.length - 3);  // Remove the ".js" extension.
+		if (bootstrapPath.endsWith('.bootstrap') && !this.jsBootstrapFiles.includes(bootstrapPath)) {
+			this.jsBootstrapFiles.push(bootstrapPath);
+		}
+
+		const info = this.jsFiles[file];
+		if (this.builder.encryptJS) {
+			if (file.indexOf('/') === 0) {
+				file = path.basename(file);
+			}
+			this.builder.jsFilesEncrypted.push(file); // original name
+			file = file.replace(/\./g, '_');
+			info.dest = path.join(this.builder.buildAssetsDir, file);
+			this.builder.jsFilesToEncrypt.push(file); // encrypted name
+		}
+
+		return new Promise((resolve) => {
+			try {
+				this.builder.cli.createHook('build.ios.copyResource', this.builder, (from, to, cb) => {
+					const originalContents = fs.readFileSync(from).toString();
+
+					const r = {
+						original: originalContents,
+						contents: originalContents,
+						symbols: []
+					};
+
+					this.builder.cli.createHook('build.ios.compileJsFile', this.builder, (r, from, to, cb2) => {
+						// Read the possibly modified file contents
+						const source = r.contents;
+
+						// If the file didn't change from previous run, return early
+						const currentHash = this.generateHash(source);
+						const previousHash = this.data.contentHashes[from];
+						if (previousHash && previousHash === currentHash) {
+							this.builder.unmarkBuildDirFile(to);
+							return cb2();
+						}
+
+						this.transformAndCopy(source, from, to, () => {
+							this.data.contentHashes[from] = currentHash;
+							cb2();
+						});
+					})(r, from, to, cb);
+				})(info.src, info.dest, resolve);
+			} catch (ex) {
+				ex.message.split('\n').forEach(m => this.logger.error(m));
+				this.logger.error(ex);
+				process.exit(1);
+			}
+		});
+	}
+
+	/**
+	 * Transforms the given JavaScript source by transpiling and minifying it and then copying
+	 * it to the destination path.
+	 *
+	 * @param {String} source JavaScript source
+	 * @param {String} from Path to the file that contains the JavaScript source
+	 * @param {String} to Path where the transformed source should be saved to
+	 * @param {Function} done Callback function
+	 */
+	transformAndCopy(source, from, to, done) {
+		// DO NOT TRANSPILE CODE inside SDK's common folder. It's already transpiled!
+		const transpile = from.startsWith(this.sdkCommonFolder) ? false : this.defaultAnalyzeOptions.transpile;
```

Comment:
extract out the result of `from.startsWith(this.sdkCommonFolder)` to a const to avoid calculating it twice (my bad!)